feature/gnmi/ate_tests/telemetry_basic_check_test/telemetry_basic_check_test.go

```diff
@@ -499,19 +499,17 @@ func TestSoftwareVersion(t *testing.T) {
 		if v, ok := parent.Val(); ok {
 			got := gnmi.Get(t, dut, gnmi.OC().Component(v).Type().State())
 
-			// Arista_7280 OC component EOS has parent type Chassis
-			if dut.Model() == "DCS-7280CR3K-32D4" {
-				if got == chassisType {
-					t.Logf("Got a valid parent %v with a type %v for the component %v", v, got, os)
-				} else {
-					t.Errorf("Got a parent %v with a type %v for the component %v, want %v", v, got, os, chassisType)
-				}
-			} else {
-				if got == supervisorType {
+			switch got {
+			case supervisorType:
+				t.Logf("Got a valid parent %v with a type %v for the component %v", v, got, os)
+			case linecardType, chassisType:
+				if deviations.OSComponentParentIsChassis(dut) || deviations.OSComponentParentIsLinecard(dut) {
```

Comment:
This check isn't as strong as it could be. Note that if the expected parent type is CHASSIS but the actual type is LINECARD, this incorrectly succeeds. I suggest refactoring this to calculate the expected parent type first, i.e.

```
want := supervisorType
switch {
case deviations.OSComponentParentIsChassis(dut):
  want = chassisType
case deviations.OSComponentParentIsLinecard(dut):
  want = linecardType
}

if got != want {
  t. Errorf(...)
}
```

Reply:
Thanks Greg. Pushed the changes.